### PR TITLE
improved a couple log messages + improved testability

### PR DESCRIPTION
**Pull request by freddyrios**
- Fixed: message about missing board with custom writes enabled was logged regardless of the setting
- removed the redundant "Stale sensor detected" that happens on some cases where report that we lost connection to a board / are attempting to reconnect
- time and logging dependencies are now injected to the CommandHandler and MCUBoard. BaseSensorBox now uses these dependencies via the CommandHandler.
- the serial port connection dependency is now injected to the MCUBoard
- added IOconfFile ctor overload that does notuse the static IOconfFileLoader.Loader

### CODE REVIEW

File: CA_DataUploaderLib/ICALogger.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace CA_DataUploaderLib
+{
+    public interface ICALogger
```
Comment (LarsWithCA) [inline CA_DataUploaderLib/ICALogger.cs:5]: why not just `ILogger` ?

File: CA_DataUploaderLib/MCUBoard.cs [hunk 2]
```diff
@@ -600,15 +594,80 @@ internal void CopyTo(MCUBoard mcuBoard)
             }
         }
 
-        public class HeaderDependencies(TimeProvider timeProvider, Action<LogID, string> logInfo, Action<LogID, string> logError, Action<LogID, string, Exception> logException)
+        public class Dependencies(
+            TimeProvider timeProvider, Action<LogID, string> logInfo, Action<LogID, string> logError,
```
Comment (LarsWithCA) [inline CA_DataUploaderLib/MCUBoard.cs:598]: How about giving an ILogger instead of 4 different actions?

**Comment by LarsWithCA** [comment]
Awesome stuff 👍

File: CA_DataUploaderLib/ICALogger.cs (hunk 1 above)
Comment (freddyrios) [reply CA_DataUploaderLib/ICALogger.cs]: used Ilog to not use the same name as Microsoft's one

File: CA_DataUploaderLib/MCUBoard.cs (hunk 2 above)
Comment (freddyrios) [reply CA_DataUploaderLib/MCUBoard.cs]: done

**Review by LarsWithCA — APPROVED** [review]

**Review by LarsWithCA — APPROVED** [review]

**Pull request merged**